arch/arc/core/isr_wrapper.S

```diff
@@ -279,8 +262,21 @@ SECTION_FUNC(TEXT, _isr_demux)
 	bl sys_trace_isr_enter
 #endif
 	/* cannot be done before this point because we must be able to run C */
-	/* r0 is available to be stomped here, and exit_tickless_idle uses it */
-	exit_tickless_idle
+#if defined(CONFIG_SYS_POWER_MANAGEMENT)
+	/* exit tickless idle */
```

Comment:
Can we keep `exit_tickless_idle` macro instead of moving its code to _isr_demux?
I'd prefer to keep this logical code separation to make it more understandable.

Reply:
This part of code only used once, therefore I think it is better to not seperate it as a macro.

Reply:
No mater it's used only one - this code part has specific functionality so lets keep it logically separated and named with macro to avoid our ASM functions to become endless monolithic piece of code.

Reply:
Modified. Thanks for advice.